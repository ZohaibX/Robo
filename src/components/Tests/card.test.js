import { shallow } from "enzyme";
import React from "react";
import Card from "../card/card";
import renderer from "react-test-renderer"; // for snapshot

// we provide the mock data , the component needs
it("expect to render card components", () => {
  const mockRobots = [
    {
      id: 1,
      username: "zohaib",
      email: "111",
      name: "zohaib butt",
    },
    {
      id: 5,
      username: "Muzi",
      email: "111",
      name: "Muzi butt",
    },
    {
      id: 3,
      username: "kaka",
      email: "111",
      name: "Muzi butt",
    },
    {
      id: 2,
      username: "baby",
      email: "111",
      name: "Muzi butt",
    },
    {
      id: 4,
      username: "baby",
      email: "111",
      name: "bablu butt",
    },
  ];

  const mockSearchInput = "butt";

  // Snapshot Testing
  const tree = renderer
    .create(<Card Robots={mockRobots} searchInput={mockSearchInput} />)
    .toJSON();
  expect(tree).toMatchSnapshot();

  // expect(
  //   shallow(<Card Robots={mockRobots} searchInput={mockSearchInput} />).length
  // ).toEqual(1);
});
