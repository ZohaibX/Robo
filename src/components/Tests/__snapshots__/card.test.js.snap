// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card components 1`] = `
<div
  className=""
>
  <div
    className="bg-light-green dib br3  ma2 grow bw2 shadow-5"
  >
    <img
      alt=""
      src="https://robohash.org/1?size=200x200"
    />
    <div
      className=""
    >
      <h2>
        zohaib butt
      </h2>
      <p>
        111
      </p>
    </div>
  </div>
  <div
    className="bg-light-green dib br3  ma2 grow bw2 shadow-5"
  >
    <img
      alt=""
      src="https://robohash.org/5?size=200x200"
    />
    <div
      className=""
    >
      <h2>
        Muzi butt
      </h2>
      <p>
        111
      </p>
    </div>
  </div>
  <div
    className="bg-light-green dib br3  ma2 grow bw2 shadow-5"
  >
    <img
      alt=""
      src="https://robohash.org/3?size=200x200"
    />
    <div
      className=""
    >
      <h2>
        Muzi butt
      </h2>
      <p>
        111
      </p>
    </div>
  </div>
  <div
    className="bg-light-green dib br3  ma2 grow bw2 shadow-5"
  >
    <img
      alt=""
      src="https://robohash.org/2?size=200x200"
    />
    <div
      className=""
    >
      <h2>
        Muzi butt
      </h2>
      <p>
        111
      </p>
    </div>
  </div>
  <div
    className="bg-light-green dib br3  ma2 grow bw2 shadow-5"
  >
    <img
      alt=""
      src="https://robohash.org/4?size=200x200"
    />
    <div
      className=""
    >
      <h2>
        bablu butt
      </h2>
      <p>
        111
      </p>
    </div>
  </div>
</div>
`;
